### PR TITLE
Add status code to KikError

### PR DESCRIPTION
**Pull request by skassam21**

### CODE REVIEW

File: kik/error.py [hunk 1]
```diff
@@ -2,5 +2,13 @@ class KikError(Exception):
...
     Exception raised by all API errors.
     The exception message is set to the server's response.
+
+    :param status_code: Status code returned by the API call
+    :type status_code: int
+    :param content: Content returned by the API call
+    :type content: string
     """
-    pass
+    def __init__(self, message, status_code, content=None):
```
Comment (davidbistolas) [inline kik/error.py]: You're calling it above as though it has a default argument, but not specifying it here - shouldn't this be 

def __init__(self, message, status_code=<some_default_error_code>, content=None):

OR change the function signature? 

Functionally not an issue, but may make debugging difficult in the future?

File: kik/api.py [hunk 2]
```diff
@@ -235,7 +235,7 @@ def set_configuration(self, config):
         )
 
         if response.status_code != 200:
-            raise KikError(response.text)
+            raise KikError(response.text, status_code=response.status_code)
```
Comment (davidbistolas) [inline kik/api.py]: See below...

File: kik/api.py [hunk 3]
```diff
@@ -200,7 +200,7 @@ def get_configuration(self):
...
```
Comment (davidbistolas) [inline kik/api.py]: See below...

File: kik/api.py [hunk 4]
```diff
@@ -64,7 +64,7 @@ def send_messages(self, messages):
...
```
Comment (davidbistolas) [inline kik/api.py]: See below...

File: kik/error.py (hunk 1 above)
Comment (skassam21) [inline kik/error.py]: I think adding a default value makes sense, but don't know what the best default code would be.. I also want it to be always set 
so maybe it's best just to change how I call the function, not as a default argument?

Comment (davidbistolas) [inline kik/error.py]: Yes- the actual concern I had wasn't that there wasn't a default, it was that the call and the definition differed - which could be confusing down the road.  While I like having defaults for these things - they can also be a catchall-trap; So in this case, altering the call (and not the signature) is probably the best way to go. Agreed.

**Pull request merged**